src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -294,7 +294,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		}
 		const lastTaskInstance = this.getLastInstance(task);
 		const terminalData = lastTaskInstance ? this._activeTasks[lastTaskInstance.getMapKey()] : undefined;
-		if (terminalData && terminalData.promise && !validInstance) {
+		if (terminalData && terminalData.promise && !validInstance && !task.configurationProperties.dependsOn?.length) {
```

Comment:
Well, it does. Try this with a current build:
```
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "shell",
      "command": "echo dependent",
      "label": "dependent",
      "problemMatcher": []
    },
    {
      "type": "shell",
      "command": "echo Running top level task...;sleep 5;echo done",
      "label": "toplevel",
      "dependsOn": ["dependent"],
      "group": {
        "kind": "build",
        "isDefault": true
      },
      "problemMatcher": []
    }
  ]
}
```

Now hit Cmd-Shift-B a few times; each one will start a new task, rather than telling you the task is already running. It was already somewhat broken (#180546) before this commit; but without this specific change, this commit would have fixed #180546. Now its completely broken it...

Reply:
Also, sorry, I didn't get notified about your response until this morning for some reason...

Reply:
@meganrogge Are you sure its *in* the insiders you're testing with? In order to get that popup, `run` has to return `{ kind: TaskExecuteKind.Active, ... }`, and after the change above, it can only do that if the task has no dependents. Tasks with dependents simply ignore their instanceLimit.

If I checkout d980921bf1 (main after this was merged), and build, the problem above repros. If I remove the ` && !task.configurationProperties.dependsOn?.length` the problem goes away; and if I checkout main as of now, the problem has been fixed by #180546.